### PR TITLE
mm/mm_heap: optimize MM_XX_SHIFT define

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
1. using LOG2_CEIL to calc MM_MIN_SHIFT/MM_MAX_SHIFT at compile time.
2. define MM_MASK_BIT to optimze code when using more significant bit of member preceding in furure.
## Impact
optimize header file
## Testing
local test

### CODE REVIEW

File: include/nuttx/lib/math32.h [hunk 1]
```diff
@@ -29,6 +29,35 @@
...
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Returns one plus the index of the most significant 1-bit of n,
+ * or if n is zero, returns zero.
+ */
+
+#if UINTPTR_MAX > UINT32_MAX
+#define FLS(n) ((n) & 0xffffffff00000000ull ? 32 + \
```
Comment (gustavonihei) [inline include/nuttx/lib/math32.h:41]:
```suggestion
#define FLS(n) ((n) & UINT64_C(0xffffffff00000000) ? 32 + \
```
Improve portability by using the conversion macros from `inttypes.h`.
For 64-bit architectures (e.g. RV64) the `ul` should be sufficient.

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/lib/math32.h (hunk 1 above)
Comment (pkarashchenko) [reply include/nuttx/lib/math32.h:41]: yeah, but many places will need such update. `ull` should work for RV64 as well

File: include/nuttx/lib/math32.h [hunk 2]
```diff
@@ -29,6 +29,35 @@
...
+                FLS32((size_t)(n) >> 32) : FLS32(n))
+#else
+#define FLS(n) FLS32(n)
```
Comment (pkarashchenko) [inline include/nuttx/lib/math32.h:44]:
```suggestion
#  define FLS(n) FLS32(n)
```

File: include/nuttx/lib/math32.h (hunk 1 above)
Comment (pkarashchenko) [inline include/nuttx/lib/math32.h:41]:
```suggestion
#  define FLS(n) ((n) & 0xffffffff00000000ull ? 32 + \
```

Comment (Donny9) [reply include/nuttx/lib/math32.h]: ok, thank you.

File: include/nuttx/lib/math32.h (hunk 2 above)
Comment (Donny9) [reply include/nuttx/lib/math32.h]: Done!

File: include/nuttx/lib/math32.h (hunk 1 above)
Comment (Donny9) [reply include/nuttx/lib/math32.h]: Done!

**Review by xiaoxiang781216 — APPROVED** [review]

File: mm/mm_heap/mm.h [hunk 3]
```diff
@@ -60,38 +61,14 @@
  *   minor performance losses.
  */
 
+#define MM_MIN_SHIFT      (LOG2_CEIL(sizeof(struct mm_freenode_s)))
```
Comment (pkarashchenko) [inline mm/mm_heap/mm.h:64]:
```suggestion
#define MM_MIN_SHIFT      LOG2_CEIL(sizeof(struct mm_freenode_s))
```

**Review by pkarashchenko — APPROVED** [review]

File: mm/mm_heap/mm.h (hunk 3 above)
Comment (Donny9) [reply mm/mm_heap/mm.h]: Done!

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**